types/vote.go

```diff
@@ -27,7 +27,7 @@ var (
 	ErrVoteInvalidBlockHash          = errors.New("invalid block hash")
 	ErrVoteNonDeterministicSignature = errors.New("non-deterministic signature")
 	ErrVoteNil                       = errors.New("nil vote")
-	ErrVoteInvalidExtension          = errors.New("invalid vote extension")
+	ErrVoteExtensionAbsent           = errors.New("vote extension absent")
```

Comment:
I got rid of it mostly because it was unused. We don't assert against it or perform any conditional logic as a result of this type. I figured I'd remove it on that grounds and instead add an error value that the callers actually use. No objections really to adding it back, but without it being used I'm not sure I see the point in keeping it.

Reply:
That makes sense. I don't have a strong desire to keep it, it just stuck out when I was reading through.